codebase/superdarn/src.lib/tk/fitmultbsid.1.0/src/write_binary.c

```diff
@@ -0,0 +1,380 @@
+		      struct FitBSIDBeam bm)
+{
+  short int has_fov;
+  int igood, irg, ngood;
+
+  int cpid, bmnum, yr, mo, dy, hr, mt, sc, us;
+  int nave, frang, rsep, rxrise, freq, noise, atten, channel, nrang;
+  float bmazm;
+  double seconds;
+
+  int16 *slist=NULL, *gsct=NULL, *gflg=NULL, *med_gflg=NULL;
+  int16 *fov=NULL, *fov_past=NULL, *grpflg=NULL, *grpnum=NULL;
+  float *pwr0=NULL, *pwr0_e=NULL, *v=NULL, *v_e=NULL;
+  float *p_l=NULL, *p_l_e=NULL, *w_l=NULL, *w_l_e=NULL;
+  float *phi0=NULL, *phi0_e=NULL, *elv=NULL, *elv_low=NULL, *elv_high=NULL;
+  float *med_pwr0=NULL, *med_pwr0_e=NULL, *med_v=NULL, *med_v_e=NULL;
+  float *med_p_l=NULL, *med_p_l_e=NULL, *med_w_l=NULL, *med_w_l_e=NULL;
+  float *med_phi0=NULL, *med_phi0_e=NULL, *med_elv=NULL; 
+  float *vh=NULL, *vh_e=NULL, *hop=NULL, *dist=NULL;
+  char *sct=NULL, **grpid=NULL, **vh_m=NULL, **region=NULL;
+
+  struct FitElv bm_elv;
+  struct CellBSIDLoc bm_loc;
+
+  /* Add the general beam data */
+  cpid = bm.cpid;
+  bmnum = bm.bm;
+  bmazm = bm.bmazm;
+
+  TimeEpochToYMDHMS(bm.time, &yr, &mo, &dy, &hr, &mt, &seconds);
+  sc = bm.intt.sc;
+  us = bm.intt.us;
+
+  TimeEpochToYMDHMS(bm.time, &yr, &mo, &dy, &hr, &mt, &seconds);
+  DataMapStoreScalar(ptr, "cpid", DATAINT, &cpid);
+  DataMapStoreScalar(ptr, "bmnum", DATASHORT, &bmnum);
+  DataMapStoreScalar(ptr, "bmazm", DATAFLOAT, &bmazm);
+  DataMapStoreScalar(ptr, "yr", DATAINT, &yr);
+  DataMapStoreScalar(ptr, "mo", DATAINT, &mo);
+  DataMapStoreScalar(ptr, "dy", DATAINT, &dy);
+  DataMapStoreScalar(ptr, "hr", DATAINT, &hr);
+  DataMapStoreScalar(ptr, "mt", DATAINT, &mt);
+  DataMapStoreScalar(ptr, "sc", DATAINT, &sc);
+  DataMapStoreScalar(ptr, "us", DATAINT, &us);
+
+  /* Add the beam parameter values */
+  nave    = bm.nave;
+  frang   = bm.frang;
+  rsep    = bm.rsep;
+  rxrise  = bm.rxrise;
+  freq    = bm.freq;
+  noise   = bm.noise;
+  atten   = bm.atten;
+  channel = bm.channel;
+  nrang   = bm.nrang;
+
+  DataMapStoreScalar(ptr, "nave", DATAINT, &nave);
+  DataMapStoreScalar(ptr, "frang", DATAINT, &frang);
+  DataMapStoreScalar(ptr, "rsep", DATAINT, &rsep);
+  DataMapStoreScalar(ptr, "rxrise", DATAINT, &rxrise);
+  DataMapStoreScalar(ptr, "freq", DATAINT, &freq);
+  DataMapStoreScalar(ptr, "noise", DATAINT, &noise);
+  DataMapStoreScalar(ptr, "atten", DATAINT, &atten);
+  DataMapStoreScalar(ptr, "channel", DATAINT, &channel);
+  DataMapStoreScalar(ptr, "nrang", DATAINT, &nrang);
+
+  /* Get the size of the data arrays (only as long as the actual data) */
+  sct = DataMapStoreArray(ptr, "sct", DATACHAR, 1, &nrang, NULL);
+
+  for(ngood = 0, irg = 0; irg < nrang; irg++)
+    {
+      sct[irg] = bm.sct[irg];
+      if(bm.sct[irg] == 1) ngood++;
+    }
+
+  /* Initiallize the range gate information */
+  slist    = DataMapStoreArray(ptr, "slist", DATASHORT, 1, &ngood, NULL);
+  gsct     = DataMapStoreArray(ptr, "gsct", DATASHORT, 1, &ngood, NULL);
+  gflg     = DataMapStoreArray(ptr, "gflg", DATASHORT, 1, &ngood, NULL);
```

Comment:
New groundscatter flag provided alongside original.  Original keeps the name, new flag has new name.